flask/helpers.py

```diff
@@ -504,6 +504,11 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
         a :class:`~datetime.datetime` or timestamp.
         If a file was passed, this overrides its mtime.
     """
+    import warnings
+    if type(filename_or_fp) is str:
```

Comment:
This is not the proper way of checking whether something is a string. Use `isinstance(.., string_types)` (assuming you want to handle both unicode and byte strings).